src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/ExtendedFortisEvent.scala

```diff
@@ -0,0 +1,21 @@
+package com.microsoft.partnercatalyst.fortis.spark.analyzer
+
+import java.util.UUID
+
+import com.microsoft.partnercatalyst.fortis.spark.dto.{Analysis, Details, FortisEvent, Location}
+
+case class ExtendedFortisEvent[T](
+  details: ExtendedDetails[T],
+  analysis: Analysis
+) extends FortisEvent
+
+case class ExtendedDetails[T](
+  id: UUID,
+  createdAtEpoch: Long,
+  body: String,
+  title: String,
+  publisher: String,
+  sourceUrl: String,
+  sharedLocations: List[Location] = List(),
+  original: T
```

Comment:
Not sure what you mean by hairy given that it's 90 lines of code with about 50 functional lines. There are many potential eventualities that we can't account for. As such, it's usually wise to defer generalization until the requirement actually exists.

My main concern here is that we spent all this time shuffling code around instead of getting in and making the minimum necessary change to get our required features built out. We're up against some pretty tight deadlines so it'd be great to avoid big changes in the future and implement what we need with the minimal amount of investment... here, unfortunately the time is already spent and I understand you're convinced that this is the right way forward, so there is little value in continuing to discuss this.